autogen/main.tf.tmpl

```diff
@@ -125,6 +125,14 @@ resource "google_compute_backend_service" "default" {
     sample_rate = lookup(lookup(each.value, "log_config", {}), "sample_rate", "1.0")
   }
 
+  dynamic "iap" {
+    for_each = var.iap_enabled ? [1] : []
+    content {
+      oauth2_client_id     = var.iap_oauth2_client_id
+      oauth2_client_secret = var.iap_oauth2_client_secret
+    }
+  }
+
```

Comment:
Could we move the vars to control this to the backend object so we can set it per backend rather then globally on the LB?

Reply:
I'm not sure, what you mean: put the dynamic block inside the dynamic "backend" block?

Reply:
Can one load balancer have more than one IAP configuration?

Reply:
That should be possible, it's set on a backend level and a single LB can consist of multiple backends. 

You can't have different IAP configs on a single backend, but you can easily create new backends pointing to the same thing with slightly different configs e.g. IAP/Caching/Timeouts

Reply:
Ok, I'll give it a try.

Reply:
Not actually tested this but to avoid adding too many object fields you could do something like this and then only add those 2 vars to the object (rather then also adding `iap_enabled`)

```tf
  dynamic "iap" {
    for_each = (lookup(each.value, "oauth2_client_id", "") != "" && lookup(each.value, "oauth2_client_secret", "") != "" ) ? [
      {
        oauth2_client_id     = each.value.oauth2_client_id
        oauth2_client_secret = each.value.oauth2_client_secret
      }
    ] : []

    content {
      oauth2_client_id     = iap.value.oauth2_client_id
      oauth2_client_secret = iap.value.oauth2_client_secret
    }
  }
```

Reply:
With this solution I got

The given value is not suitable for child module variable "backends" defined
at ../../terraform-google-lb-http/variables.tf:69,1-20: element "default":
attributes "oauth2_client_id" and "oauth2_client_secret" are required.

This one works:
```
  dynamic "iap" {
    for_each = lookup(lookup(each.value, "iap_config", {}), "enable", false) ? [1] : []
    content {
      oauth2_client_id     = lookup(lookup(each.value, "iap_config", {}), "oauth2_client_id", "")
      oauth2_client_secret = lookup(lookup(each.value, "iap_config", {}), "oauth2_client_secret", "")
    }
  }
````